mod/consensus/pkg/cometbft/service/abci.go

```diff
@@ -447,7 +417,7 @@ func (s *Service[LoggerT]) Commit(
 	context.Context, *cmtabci.CommitRequest,
 ) (*cmtabci.CommitResponse, error) {
 	if s.finalizeBlockState == nil {
-		return nil, errors.New("finalizeBlockState is nil")
+		panic(fmt.Errorf("commit: %w", errNilFinalizeBlockState))
```

Comment:
CometBFT calls ordering guarantees that this cannot happen. CosmosSDK baseApp, which is our model here, does not do this check, but we need to appease nilaway.
Turning this into a panic, as we do with `workingHash`, since I believe it's a similar situation